src/transaction_builder.js

```diff
@@ -392,14 +399,15 @@ function build (type, input, allowIncomplete) {
 
       return payments.p2pk({ signature: signatures[0] })
     }
-    case SCRIPT_TYPES.MULTISIG: {
+    case SCRIPT_TYPES.P2MS: {
+      const m = input.maxSignatures
       if (allowIncomplete) {
         signatures = signatures.map(x => x || ops.OP_0)
       } else {
         signatures = signatures.filter(x => x)
       }
-
-      return payments.p2ms({ signatures }, { allowIncomplete })
+      const validate = !allowIncomplete || (m === signatures.length)
+      return payments.p2ms({ m, pubkeys, signatures }, { allowIncomplete, validate })
```

Comment:
If the transaction is "not not complete === complete",  or if `m === signatures.length` validate!
Otherwise the `OP_0` counts being different to `m` causes this to explode...

Maybe this comment should be an inline comment in the code...

Reply:
Related https://www.google.com/search?q=OP_0+multisig+incomplete

https://github.com/bitcoinjs/bitcoinjs-lib/issues/404

Reply:
Added comment in https://github.com/bitcoinjs/bitcoinjs-lib/pull/1201/commits/7c3d1f292d6914c355e5b3e8cbd5ea732bb735cd

Reply:
Looking back at this,  maybe `payments.p2ms` with `{ allowIncomplete: true, ... }` should be what is doing the special casing, not `TransactionBuilder`... thoughts @junderw?

Shall we make an issue to track this?